dart/lib/src/enricher/io_enricher_event_processor.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:async';
+import 'dart:io';
+
+import '../protocol.dart';
+import '../sentry_options.dart';
+
+EventProcessor enricherEventProcessor(SentryOptions options) {
+  final processor = IoEnricherEventProcessor(
+    options.platformChecker.hasNativeIntegration,
+    options.sendDefaultPii,
```

Comment:
this will probably cause a bug I guess, if `sendDefaultPii` is false by default and you change it in the configuration callback, at this point you passed the boolean as false to the processor as a value copy.
maybe just pass the raw options to IoEnricherEventProcessor